test/tests/server_connectorTest.js

```diff
@@ -2945,42 +2945,11 @@ describe("Connector Server", function () {
 			let item = win.ZoteroPane.getSelectedItems()[0];
 			assert.equal(item.getDisplayTitle(), "Title!");
 
-			// Cancel the session
-			let cancelReq = await Zotero.HTTP.request(
-				'POST',
-				connectorServerPath + '/connector/cancel',
-				{
-					headers: { 'content-type': 'application/json' },
-					body: JSON.stringify({ sessionID })
-				}
-			);
-			assert.equal(cancelReq.status, 200);
-
-			// Cancelled item immediately moved to trash
-			let cancelledItem = await Zotero.Items.getAsync(item.id);
-			assert.isTrue(cancelledItem.deleted);
-			
-			// Wait for everything to finish
-			let isDone = false;
-			do {
-				await Zotero.Promise.delay(100);
-				let checkProgress = await Zotero.HTTP.request(
-					'POST',
-					connectorServerPath + "/connector/sessionProgress",
-					{
-						headers: {
-							"Content-Type": "application/json"
-						},
-						body: JSON.stringify({ sessionID })
-					}
-				);
-				let data = JSON.parse(checkProgress.response);
-				isDone = data.done;
-			}
-			while (!isDone);
+			// When /connector/cancel is called and save session is not done,
+			// it will call eraseItemsWhenDone to wait for items to load before deleting them
+			let session = Zotero.Server.Connector.SessionManager.get(sessionID);
+			await session.eraseItemsWhenDone();
```

Comment:
You should still request `/connector/cancel` here, but do something like

```js
let eraseItemsWhenDone = sinon.spy(session, 'eraseItemsWhenDone')
// ... do stuff
await eraseItemsWhenDoen.firstCall.returnValue;
```

Reply:
Oh, I actually didn't know about `sinon.spy` and didn't realize that's what you meant. Apologies! I'll keep it in mind for the future.

But it's super handy! I brought back the `/connector/cancel` call followed by `await eraseItemsWhenDoneSpy.returnValues[0]`. In addition, I thought it's worth to also spy for `session.trashItems` so we can `await` for when we move the item to the trash to make sure it does go away from the current view.